addons/crm_livechat/models/crm_lead.py

```diff
@@ -0,0 +1,28 @@
+from ast import literal_eval
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    channel_id = fields.Many2one(
```

Comment:
Make it readonly so that people are not tempted to change it in UI. That's not solving any security concern (both should be implemented) but that's some basics (anyway this field is there for information purpose mainly).